modules/percolator/src/main/java/org/elasticsearch/percolator/PercolateQueryBuilder.java

```diff
@@ -130,12 +130,12 @@
     private final Supplier<BytesReference> documentSupplier;
 
     /**
-     * @deprecated use {@link #PercolateQueryBuilder(String, BytesReference, XContentType)} with the document content
+     * @deprecated use {@link #PercolateQueryBuilder(String, String, BytesReference, XContentType)} with the document content
      * type to avoid autodetection.
      */
     @Deprecated
-    public PercolateQueryBuilder(String field, String documentType, BytesReference document) {
-        this(field, documentType, Collections.singletonList(document), XContentHelper.xContentType(document));
+    public PercolateQueryBuilder(String field, String documentType, String name, BytesReference document) {
```

Comment:
`name` is not a mandatory argument so I don't think it should be added in the ctr. Could we set it on the rewritten query instead if the value is not null ?

Reply:
updated as you suggested